### PR TITLE
Removed isAriaListbox flag, fixes #210

### LINKED ISSUE #210: List: easy way to switch between selection mode and action mode
Switching lists between selection mode and action mode is a common UI paradigm.   Ex: click the "Edit" button on the list of messages in iOS's mail app.  Or the delete button on Android's mail app (when looking at your list of messages).

IIUC, it's possible to do this with List by switching `selectionMode` from `none` to `single` (or `multiple` or `radio`).   But it doesn't seem to mesh well with the new `isAriaListbox` flag.

What is the procedure for switching a list between selection mode and action mode when using that `isAriaListBox` flag?   Is it:

```js
myListBox.isAriaListBox = false;
myListBox.selectionMode = "single";
```

and then to switch back:

```js
myListBox.selectionMode = "none";
myListBox.isAriaListBox = true;
```

(And if so, does the order of the two commands matter?)

### PR DESCRIPTION
**Pull request by tkrugg**
- Removed isAriaListbox flag
- updated doc and samples

I set up a [mirror branch](https://travis-ci.org/tkrugg/deliteful/builds/44578571) with my travis credentials where this builds successfully.

### CODE REVIEW

File: list/List.js [hunk 1]
```diff
@@ -132,16 +132,14 @@ define([
...
-		 * * The `itemRenderer` and `categoryRenderer` widget are not allowed to provide internal navigation.
-		 * @member {boolean}
-		 * @default false
-		 */
-		isAriaListbox: false,
+		setAttribute: dcl.superCall(function (sup) {
+			return function (attr, value) {
+				sup.apply(this, arguments);
+				if (attr === "role") {
+					this._propagateRole(value);
```
Comment (cjolif) [inline list/List.js]: a more common naming than "propagate" might be "_applyRole"  (shorter as well).

File: list/List.js [hunk 2]
```diff
@@ -227,9 +225,14 @@ define([
 		 * @private
 		 */
 
-		render: function () {
+		postRender: function () {
```
Comment (cjolif) [inline list/List.js]: can you remind me why you had to change this from render to postRender?

File: list/List.js [hunk 3]
```diff
@@ -467,13 +467,14 @@ define([
...
 		/*jshint maxcomplexity:12*/
-		_refreshAriaListboxProperty: function () {
-			this.setAttribute("role", this.isAriaListbox ? "listbox" : "grid");
-			if (this.isAriaListbox) {
+		_propagateRole: function (role) {
+			if (role === "listbox") {
 				if (this.selectionMode === "none") {
 					this.selectionMode = "single";
 				}
 				// TODO: SHOULD WE REMOVE THE FOLLOWING CODE FOR OPTIMIZATION ?
```
Comment (cjolif) [inline list/List.js]: I think we said we finally don't want to remove this? As switching is common so maybe we should remove that comment?

**Comment by cjolif** [comment]
@tkrugg any feedback on my comments?

File: list/List.js (hunk 1 above)
Comment (tkrugg) [inline list/List.js]: ok, I'll make that change.

File: list/List.js (hunk 3 above)
Comment (tkrugg) [inline list/List.js]: ok, i'll remove the todo

File: list/List.js (hunk 2 above)
Comment (tkrugg) [inline list/List.js]: inside _propagateRole, there is a `this.selectionMode = "single";`, and then inside refreshRendering there is a bunch of things that need to be triggered once this prop has changed. 
In order for selectionMode to be part of the next refresh cycle it needs to be outside render.
@wkeese am I doing something wrong?

Comment (wkeese) [inline list/List.js]: I think you are right, because delite/Widget doesn't call `initializeInvalidating()` until `postRender()`.

**Comment by wkeese** [comment]
FYI, I looked over this PR briefly and it seems good.

**Comment by tkrugg** [comment]
I did the changes asked.
It's not building successfully (see [log](https://travis-ci.org/tkrugg/deliteful/builds/47537215)) but I suspect it's not related to this PR, as I *tend* to get the same tests failing with the master branch.

**Comment by tkrugg** [comment]
after offline discussion with @cjolif, there is still work to be done here: I haven't completely unlinked the selectionMode and role states: as you can see it by [theses few lines](https://github.com/tkrugg/deliteful/blob/List/getting-rid-of-isAriaListbox%23210/list/List.js#L476-L478). I'm going to fix it.

**Comment by cjolif** [comment]
> I tend to get the same tests failing with the master branch.

Yes the list-prog-5 test is failing without those. I'm working on fixing the test. There should be a commit pretty soon (the problem seems to be in the tests).

**Comment by tkrugg** [comment]
This is ready. It's still [failing on list-prog-5.html](https://travis-ci.org/tkrugg/deliteful/builds/47638428) but as you can see this PR doesn't bring any change to it so there is a chance it won't conflict with the fix @cjolif is working on.

**Comment by cjolif** [comment]
merged in 1924b1e7696afcb4da4ffdd1a55fea14c59bd6f2

**Pull request closed without merging**